guard/guard.go

```diff
@@ -29,7 +29,7 @@ type PackageManagerGuardInteraction struct {
 	// Block is called to block the installation of the malware packages. One or more malicious
 	// packages are passed as arguments. These are the packages that were detected as malicious.
 	// Client code must perform the necessary error handling and termination of the process.
-	Block func(...*analyzer.PackageVersionAnalysisResult) error
+	Block func(bool, ...*analyzer.PackageVersionAnalysisResult) error
```

Comment:
I think this `bool` param is adhoc and misleading. What if you need another config in bool in future? I suggest we introduce a struct calld `BlockConfig` which currently contains a single `bool`. Name the param with appropriate intention revealing name e.g. `showReference` with appropriate comment so that we know the purpose of the param. Using a `BlockConfig` makes this scalable ie. you can add new config params without breaking the function signature.

Always introduce a `DefaultBlockConfig()` function which contains are opinionated default. The caller can override the individual config as required.